dotCMS/src/integration-test/java/com/dotmarketing/portlets/variant/business/VariantFactoryTest.java

```diff
@@ -0,0 +1,411 @@
+
+        FactoryLocator.getVariantFactory().update(variantUpdated);
+
+        final Variant variantFromDataBase = getVariantFromDataBase(variantSaved);
+
+        assertEquals(variantUpdated.getName(), variantFromDataBase.getName());
+        assertEquals(variantUpdated.getIdentifier(), variantFromDataBase.getIdentifier());
+        assertFalse(variantFromDataBase.isArchived());
+    }
+
+    /**
+     * Method to test: {@link VariantFactory#update(Variant)}
+     * When: Try to update the {@link Variant}'s deleted attribute
+     * Should: Update it in Data base.
+     *
+     * @throws DotDataException
+     */
+    @Test
+    public void updateDeletedField() throws DotDataException {
+        final Variant variant = new VariantDataGen().next();
+
+        final Variant variantSaved = FactoryLocator.getVariantFactory().save(variant);
+
+        assertNotNull(variantSaved);
+        assertNotNull(variantSaved.getIdentifier());
+        assertFalse(variantSaved.isArchived());
+
+        final Variant variantUpdated = new Variant(variantSaved.getIdentifier(),
+                variantSaved.getName(), true);
+
+        FactoryLocator.getVariantFactory().update(variantUpdated);
+
+        final Variant variantFromDataBase = getVariantFromDataBase(variantSaved);
+
+        assertEquals(variantUpdated.getName(), variantFromDataBase.getName());
+        assertEquals(variantUpdated.getIdentifier(), variantFromDataBase.getIdentifier());
+        assertTrue(variantFromDataBase.isArchived());
+    }
+
+    /**
+     * Method to test: {@link VariantFactory#update(Variant)}
+     * When: Try to update a {@link Variant} object with a Null name
+     * Should: throw a {@link NullPointerException}
+     *
+     * @throws DotDataException
+     */
+    @Test()
+    public void updateWithoutName() throws DotDataException {
+        final Variant variant = new VariantDataGen().next();
+
+        final VariantFactory variantFactory = new VariantFactoryImpl();
+        final Variant variantSaved = variantFactory.save(variant);
+
+        assertNotNull(variantSaved);
+        assertNotNull(variantSaved.getIdentifier());
+
+        final Variant variantUpdated = new Variant(variantSaved.getIdentifier(), null, false);
+
+        try {
+            variantFactory.update(variantUpdated);
+            throw new AssertionError("DotDataException Expected");
+        }catch (DotDataException e) {
+            if (DbConnectionFactory.isPostgres()) {
+                assertEquals(PSQLException.class, e.getCause().getClass());
+            } else if (DbConnectionFactory.isMsSql()){
+                assertEquals(SQLServerException.class, e.getCause().getClass());
+            } else {
+                throw new AssertionError("Database not expected");
+            }
+        }
+    }
+
+    /**
+     * Method to test: {@link VariantFactory#update(Variant)}
+     * When: Try to update a {@link Variant} object with a Null ID
+     * Should: throw a {@link NullPointerException}
```

Comment:
We don't need this test anymore because We are using Immutable now and it does not allow to create the Variant with a null name

https://github.com/dotCMS/core/commit/8aaaf9e8f7d06c71edfd820737a9fcabc12d94ed#diff-5e743d42b5d325d8137e5e4a8f2efe1b0d237033cba6cfa4f66ec1b30d7e283cL239